src/geodata2.erl

```diff
@@ -33,36 +43,57 @@ start_link(Name) ->
 stop() ->
     application:stop(geodata2).
 
-new(File) ->
-    case filelib:is_file(File) of
-        true ->
-            {ok, RawData} = file:read_file(File),
-            Data =
-                case is_compressed(File) of
-                    true ->
-                        zlib:gunzip(RawData);
-                    false ->
-                        RawData
-                end,
-            {ok, Meta} = geodata2_format:meta(Data),
-            %% @TODO [RTI-8302] This one could be removed after the IPv4+IPv6 MMDB is definitely used
-            set_is_ipv6_mmdb(Meta),
-            ets:new(?GEODATA2_STATE_TID, [set, protected, named_table, {read_concurrency, true}]),
-            ets:insert(?GEODATA2_STATE_TID, {data, Data}),
-            ets:insert(?GEODATA2_STATE_TID, {meta, Meta}),
-            {ok, #state{}};
-        _ ->
-            {stop, {geodata2_dbfile_not_found, File}}
-    end.
+new(MagellanFile, DomainFile) ->
+    MagellanState = case filelib:is_file(MagellanFile) of
+      true ->
+          {ok, RawData} = file:read_file(MagellanFile),
+          Data = case is_compressed(MagellanFile) of
+                   true ->
+                       zlib:gunzip(RawData);
+                   false ->
+                       RawData
+                 end,
+          {ok, Meta} = geodata2_format:meta(Data),
+          ets:new(?GEODATA2_STATE_TID, [set, protected, named_table, {read_concurrency, true}]),
+          ets:insert(?GEODATA2_STATE_TID, {data, Data}),
+          ets:insert(?GEODATA2_STATE_TID, {meta, Meta}),
+          {ok, #state{}};
+      _ ->
+          {stop, {geodata2_dbfile_not_found, MagellanFile}}
+    end,
+    DomainState = case filelib:is_file(DomainFile) of
```

Comment:
this looks on the right track but I see a bit of repeated code and the funciotn ended up a bit big. Id suggest to move the code from line 47 to 62 to a helper function that can be called twice, each with the right MagellanFile, DomainFile, and Tid. Then from this new function call each of them. I